src/render/posts/does-tdd-work.html.md

```diff
@@ -0,0 +1,228 @@
+size, and testing.
+
+This did appear to be the conclusion based on the data but this is a while back now so it's difficult to say if this is as true today. Also the experiments were on such small groups that it's hard to take this as a fact. To say the results are statistically significant could be a bit of a stretch but it's certainly better than no data.
+
+>This research revealed a number of differences between TDD acceptance and
+efficacy in beginning and mature developers. 
+
+This is just an interesting observation to me. Why do more experienced developers prefer TDD to newer developers? 
+
+A similar work: [Overview of the Test Driven Development
+Research Projects and Experiments ](http://proceedings.informingscience.org/InSITE2012/InSITE12p165-187Bulajic0052.pdf) also looks at varied research papers. This one is slightly newer (2012) and included some more recent research. A significant addition is a study with IBM and Microsoft development teams.
+
+Final conclusion in this study was (Nagappan et al., 2008):
+* Reducing of defect density (IBM 40%, Microsoft 60% - 90%)
+* Increase of time taken to code feature (15% - 35%).
+
+Threats to validity of the study were identified as (Nagappan et al., 2008):
+* Higher motivation of developers that were using TDD methodology.
+* The project developed by using TDD might be easier. 
+
+The conclusion of that paper were less positive than Janzen's as they rightly concluded the results were too varied and sample sizes too small to draw any positive conclusions.
+
+Another [controlled experiment conducted in 2012](http://www.ipr.mdh.se/pdf_publications/2345.pdf) once again concluded that TDD is probably a good thing but that more evidence is needed.
+
+[Another study from 2016](http://people.brunel.ac.uk/~csstmms/FucciEtAl_ESEM2016.pdf) looked at the effects of TDD compared to Test Last Development (TLD).
+
+>In this paper we reported a replication of an experiment
+[13] in which TDD was compared to a test-last approach.
+
+This study seemed to lean towards a verdict that TDD doesn't improve things over TLD:
+
+>Given the limitations presented in Section 5, it appears that
+TDD does not improve, nor deteriorate the participants’
+performance with respect to an iterative development technique
+in which unit tests are written after production code
+ 
+If you look at how the tests were conducted though you can see an interesting aspect of it is how programmers were to use iterative development. There was a fairly tight loop between test and production code for both TDD and TLD. Perhaps a tight iterative loop is more important than the order of test code?
+
+In my opinion this paper didn't provide enough data to give a conclusion one way or the other. It certainly didn't prove TDD was better or worse than writing all your tests after writing all your production code for a decent sized project. Also this study was focused in academia which can make it harder to relate to industry. 
```

Comment:
Why is the academic setting less relevant to industry? Shouldn't the study be trying to say something about TDD as a general practice independently of the context?